### ISSUE TITLE
Parallel errors

### ISSUE THREAD

**Issue by edmcman** [nsidc/vagrant-vsphere#139]
I am encountering new errors after pulling the commit that enabled parallel actions:

    An error occurred while executing multiple actions in parallel.
    Any errors that occurred are shown below.

    An error occurred while executing the action on the 'worker1'
    machine. Please handle this error then try again:

    undefined method `find_vm' for #<RbVmomi::VIM::Datacenter:0x007fabb81c46c8>

    An error occurred while executing the action on the 'worker13'
    machine. Please handle this error then try again:

    undefined method `wait_for_completion' for #<RbVmomi::VIM::Task:0x000000027d1cc8>

This error occurred while executing `vagrant up` that brought up ~5 VMs.

**Comment by marnovdm** [comment]
A colleague of mine had the same issue after I just upgraded to the latest vagrant-vsphere plugin version, however my own Vagrant projects work fine using this version, so it doesn't seem to be an issue with all projects... I'm gonna look into reproducing this.

**Comment by edmcman** [comment]
Please let me know if I can do anything to help.  I will try to debug a little more, but I am new to ruby development.

**Comment by marnovdm** [comment]
So far we've been unable to reproduce it reliably, it looks like some kind of timing issue/race condition or something. I'll update here if we find out how to reproduce it.

**Comment by kikitux** [comment]
@edmcman can you share a gist of your `Vagrantifile`

**Comment by edmcman** [comment]
@kikitux https://gist.github.com/edmcman/2485137a223cca439681

**Comment by kikitux** [comment]
try something like this: (this works for me)

(1..vm_number).each do |i|
config.vm.define vm_name = "#{vm_prefix}%02d" % i do |config|
config.vm..

**Comment by edmcman** [comment]
Hi @kikitux 

Thanks for the suggestion.  Unfortunately, this resulted in a similar error.

The problem does not exhibit itself every time... I use the following bash script to stress test:

`while vagrant up && vagrant destroy; do true; done`

**Comment by rylarson** [comment]
There is definitely some sort of race condition. I pointed this out in the PR but it was still merged. I think it should be reverted until we can get it working properly.

**Comment by edmcman** [comment]
Yeah, I agree.  I have to constantly use --no-parallel.

**Comment by marnovdm** [comment]
For me it works 99% of the time and it saves me a lot of time deploying. I do multiple deployments of 'large' (20+ vms) projects almost every day and haven't seen  the issue since my previous comment. My colleague deploys some small (2-5 vms) projects on the same vsphere infrastructure and seems to run into this more often (I'd say it works for him about 60% of the time). I haven't been able to reliably reproduce it.

**Comment by edmcman** [comment]
No offense, but saving a lot of time is not a good enough reason to break it for others.  It would be a nice feature to have, but at the moment it's simply broken.

If the patch was reverted, can't you just manually use `vagrant up --parallel`?

**Comment by marnovdm** [comment]
That wasn't my intention, just wanted to update here with my experience. Perhaps the default should indeed be changed for the time being. Not sure if --parallel would still work without the patch, haven't tested that yet.

**Comment by edmcman** [comment]
Sorry for the misunderstanding.  I think we can all agree that it would be best just to fix the problem!  I will have a little more free time at night, so I'll try to take a look at it.

**Comment by edmcman** [comment]
I tracked this down to a bug in rbvmomi.  It appears that the bug is fixed in the newest version of rbvmomi, so I am submitting a pull request to require the latest version of rbvmomi.

**Comment by rylarson** [comment]
Do you have a link to the bug?

**Comment by edmcman** [comment]
No, sorry.  After testing, I concluded the problem could only be in rbvmomi.  I then upgraded rbvmomi and the problem went away.  Unfortunately I don't know when exactly it was fixed.

**Comment by ggascoigne** [comment]
This is still broken.  I have a number of mixed windows and linux boxes that I tried bringing up (only two of each so nothing too heavy).  One time I ended out with errors about sudo not being found on my Windows boxes (from part of the provisioning scripts). Another time I got errors installing the new ssh keys on a box that didn't support it.  Basically it looks like somewhere the concurrently accessed hosts are getting mixed up.

Starting them with the --no-parallel consistently resolved the issue.

I'm fine with the option to use the parallel startup being supported, but I really wish that the change to make it the default could be rolled back.
